### ISSUE TITLE
SIGSEGV with type alias to non-checked package

### ISSUE THREAD

**Issue by ongardie-ebay** [dominikh/go-tools#379]
The problem appears to be caused by unintentionally checking code that's pulled in with a type alias. If the aliased code resides in a package outside the set of target packages, the errors found over there won't have a package available, leading to a nil pointer dereference.

This script reproduces the problem and probably explains it better than my description above. Run it in an empty directory or follow along manually:
```bash
#!/bin/bash
set -ex

export GOPATH=$(pwd)
go get -v honnef.co/go/tools/cmd/staticcheck

mkdir -p src/example.org/{dep,main}

cat > src/example.org/dep/dep.go << 'EOF'
package dep

type Foo struct{}

func (a Foo) String() string {
	return "Foo"
}

func (x Foo) Equal(y Foo) bool {
	return true
}
EOF

cat > src/example.org/main/main.go << 'EOF'
package main

import (
	"fmt"

	"example.org/dep"
)

type Bar = dep.Foo

func main() {
	fmt.Println(Bar{})
}
EOF

go install example.org/main/...

# This is OK:
./bin/staticcheck example.org/main/... example.org/dep/...

# This fails with SIGSEGV:
./bin/staticcheck example.org/main/...
```

Here's the output:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x6b2bb2]

goroutine 1 [running]:
honnef.co/go/tools/lint.(*Linter).Lint(0xc003cff970, 0xc0050c6950, 0x1, 0x1, 0xc003cff810, 0x1, 0x1, 0x0)
	/go/src/honnef.co/go/tools/lint/lint.go:473 +0x1f22
honnef.co/go/tools/lint/lintutil.Lint(0xc0000e7020, 0x4, 0x6, 0xc000092010, 0x1, 0x1, 0xc0004b9e40, 0x0, 0xc0000d1b18, 0x4d1384, ...)
	/go/src/honnef.co/go/tools/lint/lintutil/util.go:307 +0x725
honnef.co/go/tools/lint/lintutil.ProcessFlagSet(0xc0000e7020, 0x4, 0x6, 0xc0000e6fc0)
	/go/src/honnef.co/go/tools/lint/lintutil/util.go:178 +0x99d
main.main()
	/go/src/honnef.co/go/tools/cmd/staticcheck/staticcheck.go:29 +0x2cb
```

**Comment by dominikh** [comment]
Fixed on master; will be part of 2019.1.1

**Comment by ongardie-ebay** [comment]
Thanks @dominikh.
